src/DtlsServer.cc

```diff
@@ -59,38 +79,51 @@ DtlsServer::DtlsServer(const Napi::CallbackInfo& info) : Napi::ObjectWrap<DtlsSe
 	mbedtls_entropy_init(&entropy);
 	mbedtls_ctr_drbg_init(&ctr_drbg);
 
-	// Required 1st param: key
-	if (info.Length() < 1 || !info[0].IsBuffer()) {
-		Napi::TypeError::New(env, "Expecting first parameter (key) to be a buffer").ThrowAsJavaScriptException();
-		return;
+	// Optional 1st param: key
+	if (info.Length() >= 1 && info[0].IsBuffer()) {
```

Comment:
what happens if no parameters are passed in? I see previously it would throw an exception. Is there any instance that we would want to throw an exception based on incorrect parameters being passed in or if there were no parameters passed in?

Reply:
If no key is passed in (as we do not pass in a key), then we skip the parts of the constructor that set up the private key.
This is different from the psk (pre-shared key).  The key being specified here is the private key for a certificate.  Since we do not provide a certificate, we should not specify a private key for it.  The Particle IoT upstream version requires specifying both a certificate and its private key, but we do not, and the base library, mbedtls, does not require it.